vector/src/main/java/im/vector/app/features/notifications/NotificationBitmapLoader.kt

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2019 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.notifications
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.os.Build
+import androidx.annotation.WorkerThread
+import androidx.core.graphics.drawable.IconCompat
+import com.bumptech.glide.Glide
+import com.bumptech.glide.load.DecodeFormat
+import com.bumptech.glide.load.resource.bitmap.CircleCrop
+import com.bumptech.glide.signature.ObjectKey
+import timber.log.Timber
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class NotificationBitmapLoader @Inject constructor(private val context: Context) {
+
+    /**
+     * Get icon of a room
+     */
+    @WorkerThread
+    fun getRoomBitmap(path: String?): Bitmap? {
+        if (path == null) {
+            return null
+        }
+        return loadRoomBitmap(path)
+    }
+
+    @WorkerThread
+    private fun loadRoomBitmap(path: String): Bitmap? {
+        return try {
```

Comment:
the previous in memory caches have been removed, they were repeating the glide in memory cache and potentially causing leaks (as we never cleaned up the references)